point_viewer_proto_rust/src/proto.proto

```diff
@@ -75,23 +75,47 @@ message NodeId {
   int64 deprecated_index = 2;
 }
 
-message Node {
-  enum PositionEncoding {
+enum PositionEncoding {
     INVALID = 0;
     Uint8 = 1;
     Uint16 = 2;
     Float32 = 3;
     Float64 = 4;
-  }
+}
 
+message Node {
   PositionEncoding position_encoding = 2;
   int64 num_points = 3;
   NodeId id = 4;
 }
 
+enum AttributeDataType {
+    INVALID_DATA_TYPE = 0;
+    I64 = 1;
```

Comment:
Nit: How about numbering them according to a schema like unsigned integers = `1X`, signed integers = `2X`, floats = `3X`, vec2 = `1YX` with `Y ` like above, vec3 = `2YX`, so that when we add datatypes they're not all jumbled? You could also add all the ints from u/i8 to u/i64.

Reply:
the schema does not completely work with multiplication but with offset addition, otherwise without you get some real jam. 
enum AttributeDataType {

    INVALID_DATA_TYPE = 0;
    U8 = 1;
    **U16 = 2;**
    U32 = 3;
    U64 = 4;
    **I8 = 2*1;**
    I16 = 2*2;
    I32 = 2*3;
    I64 = 2*4;

    F32 = 3*3;
    F64 = 3*4; //max value 12
    
    U8Vec3 = 12*2 + U8 ;
    F64Vec3 = 6;
}

Reply:
I'm not sure I understand – I'm proposing that unsigned integers are 10, 11, 12, ... and signed integers are 20, 21, 22, ..., etc., U8Vec3 would then be 210.

Reply:
now i got it! I did it with just a tighter offset